### ISSUE TITLE
Expose inner glTF transforms/materials

### ISSUE THREAD

**Issue by guusw** [fragcolor-xyz/chainblocks#169]

**Comment by guusw** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Whole-Team-623c70cfcdeee40011bd03e3/issues/fragcolor-xyz/shards/169?planning-poker) with Zenhub @Kryptos-FR @sinkingsugar @tdbe

**Comment by guusw** [comment]
The way I imagined this is that you'd be able to pass the bones/subobjects that you want to modify.
Imagine the glTF looks something like
```
Root
| - Cube
| - Container
  | - Sphere
  | - Cube
```

Here's some example usages, assuming the new desired API where you can activate this every frame and it just updates the fields:

To adjust the material of all objects named `Cube`, you'd do:
```clojure
[{:Name "Cube" :Material .cube-material}] (GFX.glTF "model.gltf") ...
```

To adjust the transform of the object called `Container`, you'd do:
```clojure
[{:Name "Container" :Transform .transform}] (GFX.glTF "model.gltf") ...
```

And because the input is a sequence the operations can be combined.

**Comment by guusw** [comment]
Yeah or parameter, but it'll still be a table because you'd need to map things by name in the glTF:
```clojure
(GFX.glTF "model.gltf" :Mapping [{:Name "Container" :Transform .transform} {:Name "Cube" :Material .cube-material}]) ...

**Comment by Kryptos-FR** [comment]
Should probably allow passing a seq variable as well.

```clj
{:Name "Container" :Transform .transform} >> .mapping
{:Name "Cube" :Material .cube-material} >> .mapping
(GFX.glTF "model.gltf" :Mapping .mapping)
```

**Comment by sinkingsugar** [comment]
Closing as outdated
